bayes_search.py

```diff
@@ -145,51 +165,75 @@ def train_gaussian_process(
     return gp, y_mean, y_stddev
 
 
-def filter_weird_values(sample_X, sample_y):
+def filter_nans(sample_X: npt.ArrayLike, sample_y: npt.ArrayLike) -> npt.ArrayLike:
     is_row_finite = ~(np.isnan(sample_X).any(axis=1) | np.isnan(sample_y))
     sample_X = sample_X[is_row_finite, :]
     sample_y = sample_y[is_row_finite]
     return sample_X, sample_y
 
 
 def next_sample(
-    sample_X,
-    sample_y,
-    X_bounds=None,
-    runtimes=None,
-    failures=None,
-    current_X=None,
-    nu=1.5,
-    max_samples_for_gp=100,
-    improvement=0.01,
-    num_points_to_try=1000,
-    opt_func="expected_improvement",
-    test_X=None,
-):
+    sample_X: npt.ArrayLike,
+    sample_y: npt.ArrayLike,
+    X_bounds: Optional[npt.ArrayLike] = None,
+    runtimes: Optional[npt.ArrayLike] = None,
```

Comment:
Which of these are used, runtimes is not used?

I think there is a balance between minimizing changes to the old code and making things better.
As a reader it would help to know what parts of the code are dead/not-enabled

Does it make sense to comment out for now? i will yield to you on this, maybe just commenting certain functions or args that are not actually used.

Reply:
good point, will remove these.

Reply:
commented these out.